woocommerce/woocommerce
# Add a catch-all return to LYS hub after user exits essential task

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:






Adds a "catch-all" on the homescreen to return users to the Launch Your Store hub after completing an essential task. It works by having the homescreen check the sessionStorage key `lysWaiting`, which if present, and a task is not also being shown, will create a redirect to the lys hub. This allows tasks to continue with their existing behavior -- existing links in tasks to the homescreen will be intercepted and redirected to LYS. For tasks that don't have an endpoint at the homescreen, when users eventually navigate to the homescreen, LYS will automatically open.


https://github.com/woocommerce/woocommerce/assets/9312929/f574188a-78d3-4469-932e-064add29ad2e


Closes #47190 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Open Launch Your Store Task
2. Click on shipping task from essential tasks
3. In shipping task, press back button
4. See that LYS hub is shown
5. Click on shipping task again
6. Complete shipping task
7. See that LYS hub is shown
8. Press back button in LYS hub
9. See that homescreen is shown.









### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
